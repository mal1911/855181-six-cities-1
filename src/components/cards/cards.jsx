import React from 'react';
import PropTypes from 'prop-types';
import {offerType} from '../../prop-types';
import Card from '../card/card';
import {connect} from "react-redux";
import {getResultOffersData} from '../../reducer/data/selectors';

const Cards = (props) => {
  const handlerChange = (cardObj) => {
    props.setActiveItem(cardObj.id);
  };

  const cards = props.filteredOffersData.map((offerObj, index) =>
    <Card key={index} offerObj={offerObj} onChange={handlerChange}/>);

  return <div className="cities__places-list places__list tabs__content">
    {cards}
  </div>;
};

Cards.propTypes = {
  filteredOffersData: PropTypes.arrayOf(offerType.isRequired).isRequired,
  //activeCity: PropTypes.string.isRequired,
  activeItem: PropTypes.number.isRequired,
  setActiveItem: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  //activeCity: state.activeCity,
  //filteredOffersData: state.filteredOffersData,
  filteredOffersData: getResultOffersData(state),
});

export {Cards};
export default connect(mapStateToProps)(Cards);
