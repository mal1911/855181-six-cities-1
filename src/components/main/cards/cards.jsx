import React from "react";
import PropTypes from "prop-types";
import {offerType} from "../../../prop-types";
import {connect} from "react-redux";
import {ActionCreator} from "../../../reducer/offers-data/offers-data";
import Card from "../../card/card";

import {getResultOffersData} from "../../../reducer/offers-data/selectors";

const Cards = (props) => {
  const handleChange = (cardObj) => {
    props.onChangeActiveCard(cardObj.id);
  };

  const cards = props.resultOffersData.map((offerObj, index) =>
    <Card key={index} offerObj={offerObj} onChange={handleChange} cardClassName={`cities`}/>);

  return <div className="cities__places-list places__list tabs__content">
    {cards}
  </div>;
};

Cards.propTypes = {
  resultOffersData: PropTypes.arrayOf(offerType.isRequired).isRequired,
  onChangeActiveCard: PropTypes.func,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  resultOffersData: getResultOffersData(state),
});

const mapDispatchToProps = (dispatch) => ({
  onChangeActiveCard: (id) => {
    dispatch(ActionCreator.changeActiveOfferId(id));
  },
});

export {Cards};

export default connect(mapStateToProps, mapDispatchToProps)(Cards);
