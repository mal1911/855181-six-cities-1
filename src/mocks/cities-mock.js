export const citiesMock = [
  {
    id: 1,
    name: `Paris`,
  },
  {
    id: 2,
    name: `Cologne`,
  },
  {
    id: 3,
    name: `Brussels`,
  },
  {
    id: 4,
    name: `Amsterdam`,
  },
  {
    id: 5,
    name: `Hamburg`,
  },
  {
    id: 6,
    name: `Dusseldorf`,
  },
];

