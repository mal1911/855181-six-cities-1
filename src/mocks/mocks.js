export const offersData = [
  {
    id: 1,
    isPremium: true,
    city: `Paris`,
    title: `Paris Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`,
    coordinates: [52.3909553943508, 4.85309666406198],
  },
  {
    id: 2,
    isPremium: false,
    city: `Paris`,
    title: `Paris Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`,
    coordinates: [52.369553943508, 4.85309666406198],
  },
  {
    id: 3,
    isPremium: false,
    city: `Paris`,
    title: `Paris Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 4,
    isPremium: false,
    city: `Paris`,
    title: `Paris Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 5,
    isPremium: true,
    city: `Cologne`,
    title: `Cologne Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 6,
    isPremium: false,
    city: `Cologne`,
    title: `Cologne Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 7,
    isPremium: true,
    city: `Brussels`,
    title: `Brussels Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 8,
    isPremium: false,
    city: `Brussels`,
    title: `Brussels Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 9,
    isPremium: true,
    city: `Amsterdam`,
    title: `Amsterdam Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 10,
    isPremium: false,
    city: `Amsterdam`,
    title: `Amsterdam Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 11,
    isPremium: true,
    city: `Hamburg`,
    title: `Hamburg Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 12,
    isPremium: false,
    city: `Hamburg`,
    title: `Hamburg Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 13,
    isPremium: true,
    city: `Dusseldorf`,
    title: `Dusseldorf Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 14,
    isPremium: false,
    city: `Dusseldorf`,
    title: `Dusseldorf Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
];

export const citiesData = [
  {
    name: `Paris`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Cologne`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Brussels`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Hamburg`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Dusseldorf`,
    coordinates: [52.38333, 4.9],
  },
];
