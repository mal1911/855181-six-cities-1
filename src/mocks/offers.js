export const offers = [
  {
    title: `Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`
  },
  {
    title: `Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`
  },
  {
    title: `Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`
  },
  {
    title: `Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`
  },
];
