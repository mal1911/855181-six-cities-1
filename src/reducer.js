const initialState = {
  activeCityId: 1,
};
/*
const isArtistAnswerCorrect = (userAnswer, question) =>
  userAnswer.artist === question.song.artist;

const isGenreAnswerCorrect = (userAnswer, question) =>
  userAnswer.every((it, i) => it === (
    question.answers[i].genre === question.genre
  ));
*/

const ActionCreator = {
  changeActiveCity: (activeCityObj) => ({
    type: `CHANGE_ACTIVE_CITY`,
    payload: activeCityObj.id,
  }),

  /*
    incrementStep: () => ({
      type: `INCREMENT_STEP`,
      payload: 1,
    }),

    incrementMistake: (userAnswer, question, mistakes, maxMistakes) => {
      let answerIsCorrect = false;

      switch (question.type) {
        case `artist`:
          answerIsCorrect = isArtistAnswerCorrect(userAnswer, question);
          break;
        case `genre`:
          answerIsCorrect = isGenreAnswerCorrect(userAnswer, question);
          break;
      }

      if (!answerIsCorrect && mistakes + 1 >= maxMistakes) {
        return {
          type: `RESET`,
        };
      }

      return {
        type: `INCREMENT_MISTAKES`,
        payload: answerIsCorrect ? 0 : 1,
      };
    },*/
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `CHANGE_ACTIVE_CITY`: return Object.assign({}, state, {
      activeCityId: action.payload,
    });
/*
    case `INCREMENT_MISTAKES`: return Object.assign({}, state, {
      mistakes: state.mistakes + action.payload,
    });

    case `RESET`: return Object.assign({}, initialState);*/
  }

  return state;
};


export {
  ActionCreator,
  reducer,
};
