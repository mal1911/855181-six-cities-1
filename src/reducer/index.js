import {combineReducers} from "redux";
import {reducer as offersData} from "./offers-data/offers-data";
import {reducer as favoritesData} from "./favorites-data/favorites-data";
import {reducer as user} from "./user/user";
import NameSpace from "./name-spaces";

export default combineReducers({
  [NameSpace.OFFERS_DATA]: offersData,
  [NameSpace.FAVORITES_DATA]: favoritesData,
  [NameSpace.USER]: user,
});
