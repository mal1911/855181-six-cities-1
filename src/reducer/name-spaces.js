const NameSpace = {
  OFFERS_DATA: `OFFERS_DATA`,
  FAVORITES_DATA: `FAVORITES_DATA`,
  USER: `USER`
};

export default NameSpace;
