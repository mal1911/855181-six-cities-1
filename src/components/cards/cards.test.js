import React from 'react';
import renderer from 'react-test-renderer';
import {Cards} from './cards';
import {citiesData} from '../../mocks/mocks';
import {getFilteredOffersData} from '../../reducer';

it(`Cards correctly renders`, () => {
  const tree = renderer
    .create(<Cards
      filteredOffersData={getFilteredOffersData(citiesData[0])}
      activeItem={0}
      activeCity={citiesData[0].name}
      setActiveItem={jest.fn()}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});
