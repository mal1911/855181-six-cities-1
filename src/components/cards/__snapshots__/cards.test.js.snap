// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards correctly renders 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    0
     places to stay in 
    Paris
  </b>
  <form
    action="#"
    className="places__sorting"
    method="get"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by
    </span>
    <span
      className="places__sorting-type"
      tabIndex="0"
    >
      Popular
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlinkHref="#icon-arrow-select"
        />
      </svg>
    </span>
    <ul
      className="places__options places__options--custom places__options--opened"
    >
      <li
        className="places__option places__option--active"
        tabIndex="0"
      >
        Popular
      </li>
      <li
        className="places__option"
        tabIndex="0"
      >
        Price: low to high
      </li>
      <li
        className="places__option"
        tabIndex="0"
      >
        Price: high to low
      </li>
      <li
        className="places__option"
        tabIndex="0"
      >
        Top rated first
      </li>
    </ul>
  </form>
  <div
    className="cities__places-list places__list tabs__content"
  />
</section>
`;
